### ISSUE TITLE
Error parsing vsftpd logs when day of the month is a single digit

### ISSUE THREAD

**Issue by keliansb** [crowdsecurity/hub#526]
The parsing of vsftpd logs when the day of the month is a single digit is failing. This is because there is an extra space in the timestamp field, so the `HTTPDERROR_DATE` format does not work.

Here are syslog logs in rfc5424 format for testing purposes :
```
<13>1 2022-08-08T10:01:37.377+00:00 {"name":"test"} LOGSTASH - - - 2022-08-08T10:01:37.377Z {name=securecodebox-test} Mon Aug  8 10:01:37 2022 [pid 28368] [test] FAIL LOGIN: Client "::ffff:127.0.0.1"
<13>1 2022-08-08T10:01:37.377+00:00 {"name":"test"} LOGSTASH - - - 2022-08-08T10:01:37.377Z {name=securecodebox-test} Mon Aug 18 10:01:37 2022 [pid 28365] [test] FAIL LOGIN: Client "::ffff:127.0.0.1"
```
The fllowing output is returned with the command `cscli explain --file output.txt --type vsftpd --verbose` :
```
line: <13>1 2022-08-08T10:01:37.377+00:00 {"name":"securecodebox-test"} LOGSTASH - - - 2022-08-08T10:01:37.377Z {name=securecodebox-test} Mon Aug  8 10:01:37 2022 [pid 28368] [test] FAIL LOGIN: Client "::ffff:127.0.0.1"
        ├ s00-raw
        |       ├ 🟢 crowdsecurity/non-syslog (first_parser)
        |       └ 🔴 crowdsecurity/syslog-logs
        ├ s01-parse
        |       ├ 🔴 crowdsecurity/apache2-logs
        |       ├ 🔴 crowdsecurity/haproxy-logs
        |       ├ 🔴 crowdsecurity/nginx-logs
        |       ├ 🔴 crowdsecurity/sshd-logs
        |       └ 🔴 vsftpd-logs
        └-------- parser failure 🔴

line: <13>1 2022-08-08T10:01:37.377+00:00 {"name":"securecodebox-test"} LOGSTASH - - - 2022-08-08T10:01:37.377Z {name=securecodebox-test} Mon Aug 18 10:01:37 2022 [pid 28365] [test] FAIL LOGIN: Client "::ffff:127.0.0.1"
        ├ s00-raw
        |       ├ 🟢 crowdsecurity/non-syslog (first_parser)
        |       └ 🔴 crowdsecurity/syslog-logs
        ├ s01-parse
        |       ├ 🔴 crowdsecurity/apache2-logs
        |       ├ 🔴 crowdsecurity/haproxy-logs
        |       ├ 🔴 crowdsecurity/nginx-logs
        |       ├ 🔴 crowdsecurity/sshd-logs
        |       └ 🟢 vsftpd-logs (+7 ~2)
        |               └ update evt.Stage : s01-parse -> s02-enrich
        |               └ create evt.Parsed.source_ip : 127.0.0.1
        |               └ create evt.Parsed.timestamp : Mon Aug 18 10:01:37 2022
        |               └ create evt.Parsed.user : test
        |               └ update evt.StrTime :  -> Mon Aug 18 10:01:37 2022
        |               └ create evt.Meta.log_type : ftp_failed_auth
        |               └ create evt.Meta.program : vsftpd
        |               └ create evt.Meta.source_ip : 127.0.0.1
        |               └ create evt.Meta.user : test
        ├ s02-enrich
        |       ├ 🟢 crowdsecurity/dateparse-enrich (+2 ~1)
        |               ├ create evt.Enriched.MarshaledTime : 2022-08-18T10:01:37Z
        |               ├ update evt.MarshaledTime :  -> 2022-08-18T10:01:37Z
        |               ├ create evt.Meta.timestamp : 2022-08-18T10:01:37Z
        |       ├ 🟢 crowdsecurity/geoip-enrich (+9)
        |               ├ create evt.Enriched.ASNNumber : 0
        |               ├ create evt.Enriched.ASNOrg :
        |               ├ create evt.Enriched.ASNumber : 0
        |               ├ create evt.Enriched.IsInEU : false
        |               ├ create evt.Enriched.IsoCode :
        |               ├ create evt.Enriched.Latitude : 0.000000
        |               ├ create evt.Enriched.Longitude : 0.000000
        |               ├ create evt.Meta.ASNNumber : 0
        |               ├ create evt.Meta.IsInEU : false
        |       └ 🔴 crowdsecurity/http-logs
        ├-------- parser success 🟢
        ├ Scenarios
                └ 🟢 crowdsecurity/vsftpd-bf
```

**Comment by keliansb** [comment]
In fact I'm facing the same issue with logstash. I ended up with this solution for the timestamp field :
```
# Use separate pattern to HTTPDERROR_DATE for single-digit day-in-month ("Jul  1")
CUSTOMTIMESTAMP %{DAY} %{MONTH}  %{MONTHDAY} %{TIME} %{YEAR}
(%{HTTPDERROR_DATE:timestamp}|%{CUSTOMTIMESTAMP:timestamp}) \[pid %{NUMBER}\] ...
```

**Comment by rr404** [comment]
You could also use an optionnal space:
```
CUSTOMTIMESTAMP %{DAY} %{MONTH} (?:\s?)%{MONTHDAY} %{TIME} %{YEAR}
(%{CUSTOMTIMESTAMP:timestamp}) \[pid %{NUMBER}\] ...
```

**Comment by rr404** [comment]
We will integrate this fix as soon as the **1.4.2** is live.
The reason being that "timestamp" will sometimes get an extra space and this could be problematic for replays among other things.

_(spoiler alert)_
**1.4.2** will allow us to define a static: StrTimeFormat with a similar pattern as value, this way StrTime won't with this extra space won't be an issue.

**Comment by rr404** [comment]
We will integrate this fix as soon as the 1.4.2 is live.
The reason being that "timestamp" will sometimes get an extra space and this could be problematic for replays among other things.

(spoiler alert)
1.4.2 will allow us to define a static: StrTimeFormat with a similar pattern as value, this way StrTime with this extra space won't be problematic.
